query-request.hh

```diff
@@ -375,6 +375,11 @@ struct forward_request {
 
     db::consistency_level cl;
     lowres_clock::time_point timeout;
+
+    struct null_value {};
+    struct unset_value {};
+    std::vector<std::variant<bytes, null_value, unset_value>> bound_values;
+    std::optional<std::vector<sstring>> bound_names;
```

Comment:
I don't understand this at all. Why do you need to forward bound variables at all? Why would the intermediate coordinator need them? Aren't all the values resolved already in  `pr` and `cmd`?

Reply:
How can the intermediate coordinator see a reference to a variable?

Reply:
From what I know, the implementation of filtering needs access to bound values. Although we don't support filtering in this PR, I think it will be needed someday.

I don't quite understand your question about seeing a reference to a variable. 
If by "seeing a reference to a variable" you meant having access to `query_options` filled with bound values, then it's done in a following way:
Super-coordinator gets it's bound values+names and sends it to intermediate coordinators using `forward_request`. Upon `forward_request` receival, intermediate coordinators recreate super-coordinator's `query_options` (using the received values for this). Recreated `query_options` are then used to execute a query with `service::pager::query_pagers::pager`. So the pager has an access to the bound values via provided `query_options`.

Reply:
> From what I know, the implementation of filtering needs access to bound values. Although we don't support filtering in this PR, I think it will be needed someday.

It may, but we might substitute the variables in the expression. i.e. instead of `foo = :var`, send `foo = 17`. Or we might do some other thing.

So let's defer that until we know how we implement it, and until we can test it actually works.

> 
> I don't quite understand your question about seeing a reference to a variable. If by "seeing a reference to a variable" you meant having access to `query_options` filled with bound values, then it's done in a following way: Super-coordinator gets it's bound values+names and sends it to intermediate coordinators using `forward_request`. Upon `forward_request` receival, intermediate coordinators recreate super-coordinator's `query_options` (using the received values for this). Recreated `query_options` are then used to execute a query with `service::pager::query_pagers::pager`. So the pager has an access to the bound values via provided `query_options`.


That's the variables themselves. A reference to the variable is a lookup on the variable name or index. Today this is done by `constant evaluate(const bind_variable& bind_var, const query_options& options)`. I don't think it can be called in the forwarder service, since the query isn't passed from the supercoordinator, it is regenerated.

Reply:
Thanks for the explanation. Based on what you said, I think it would be good to remove sending bound values to immediate coordinator. Also, in a previous version there was a check that disqualified statements with bound values from a parallel execution. I think this chceck would be still unnecessary, as filtering queries are not permitted (so all bound values will be resolved on the super-coordinator side and send in `pr` and `cmd`).

So, I'll remove sending bound values, and won't restore the previously mentioned check.

Reply:
(so I agree the check should be kept removed)